File: class/class4.1/src/PizzaBuilder.cpp
#include "PizzaBuilder.hpp"

PizzaBuilder& PizzaBuilder::addIngredient(Ingredient&& ingredient) {
    this->pizza.addIngredient(std::move(ingredient));
    return *this;
}

Pizza PizzaBuilder::build() const {
    return this->pizza;
}

File: home/home4.1/pizza-builder/src/pizza_builder.rs
use crate::ingredient::Ingredient;
use crate::pizza::Pizza;

pub struct PizzaBuilder {
    pizza: Pizza,
}

impl PizzaBuilder {
    pub fn new() -> Self {
        Self {
            pizza: Pizza::new(),
        }
    }

    pub fn add_ingredient(mut self, ingredient: Ingredient) -> Self {
        self.pizza.add_ingredient(ingredient);
        self
    }

    pub fn build(self) -> Pizza {
        self.pizza
    }
}
